diff --git a/arch/arm/dts/Makefile b/arch/arm/dts/Makefile
index 3941a08cf4..7ea0dffeb5 100644
--- a/arch/arm/dts/Makefile
+++ b/arch/arm/dts/Makefile
@@ -1004,7 +1004,8 @@ dtb-$(CONFIG_TARGET_SAMA5D2_PTC_EK) += \
 	at91-sama5d2_ptc_ek.dtb
 
 dtb-$(CONFIG_TARGET_SAMA5D2_XPLAINED) += \
-	at91-sama5d2_xplained.dtb
+	at91-sama5d2_xplained.dtb \
+	at91-sama5d27_jupiter_nano.dtb
 
 dtb-$(CONFIG_TARGET_SAMA5D27_SOM1_EK) += \
 	at91-sama5d27_som1_ek.dtb   \
diff --git a/include/configs/sama5d2_xplained.h b/include/configs/sama5d2_xplained.h
index 4f5ceca780..c4bc23e4d0 100644
--- a/include/configs/sama5d2_xplained.h
+++ b/include/configs/sama5d2_xplained.h
@@ -31,9 +31,90 @@
 /* bootstrap + u-boot + env in sd card */
 #undef CONFIG_BOOTCOMMAND
 
-#define CONFIG_BOOTCOMMAND	"fatload mmc " CONFIG_ENV_FAT_DEVICE_AND_PART " 0x21000000 at91-sama5d2_xplained.dtb; " \
-				"fatload mmc " CONFIG_ENV_FAT_DEVICE_AND_PART " 0x22000000 zImage; " \
-				"bootz 0x22000000 - 0x21000000"
+#define CONFIG_SUPPORT_RAW_INITRD
+#define CONFIG_EXTRA_ENV_SETTINGS \
+    "initrd_high=0xffffffff\0" \
+    "fdt_high=0xffffffff\0" \
+    "fdtovaddr=0x21800000\0" \
+    "loadaddr=0x22000000\0" \
+    "fdtaddr=0x21000000\0" \
+    "fdtfile=/dtbs/at91-sama5d27_jupiter_nano.dtb\0" \
+    "enable_uboot_overlays=\0" \
+    "dtboverlay=\0" \
+    "dtboverlay1=\0" \
+    "dtboverlay2=\0" \
+    "dtboverlay3=\0" \
+    "dtboverlay4=\0" \
+    "console=ttyS0,115200\0" \
+    "optargs=\0" \
+    "video=\0" \
+    "mmcdev=0\0" \
+    "mmcpart=1\0" \
+    "mmcroot=/dev/mmcblk0p2 ro\0" \
+    "mmcrootfstype=ext4 rootwait\0" \
+    "mmcargs=setenv bootargs console=${console} " \
+        "${optargs} " \
+        "root=${mmcroot} " \
+        "rootfstype=${mmcrootfstype} " \
+        "video=${video}\0" \
+    "loadbootenv=load mmc ${mmcdev}:${mmcpart} ${loadaddr} uEnv.txt\0" \
+    "importbootenv=echo Importing environment from mmc (uEnv.txt)...; " \
+        "env import -t ${loadaddr} ${filesize}\0" \
+    "loadzimage=load mmc ${mmcdev}:${mmcpart} ${loadaddr} zImage\0" \
+    "loadfdt=load mmc ${mmcdev}:${mmcpart} ${fdtaddr} ${fdtfile}\0" \
+    "loadoverlay=load mmc ${mmcdev}:${mmcpart} ${fdtovaddr} ${dtboverlay}\0" \
+    "loadoverlay1=load mmc ${mmcdev}:${mmcpart} ${fdtovaddr} ${dtboverlay1}\0" \
+    "loadoverlay2=load mmc ${mmcdev}:${mmcpart} ${fdtovaddr} ${dtboverlay2}\0" \
+    "loadoverlay3=load mmc ${mmcdev}:${mmcpart} ${fdtovaddr} ${dtboverlay3}\0" \
+    "loadoverlay4=load mmc ${mmcdev}:${mmcpart} ${fdtovaddr} ${dtboverlay4}\0" \
+    "mmcboot=echo Booting from mmc ...; " \
+        "run mmcargs; " \
+        "bootz ${loadaddr} - ${fdtaddr}\0"
+
+#define CONFIG_BOOTCOMMAND \
+    "mmc dev ${mmcdev};" \
+    "if mmc rescan; then " \
+        "echo SD/MMC found on device ${mmcdev};" \
+    "fi;" \
+    "mw.w 20008000 0 100000;" \
+    "fatload mmc 0 0x20008000 nuttx.bin;" \
+    "fatload mmc 0 0x20008000 nuttx.bin;" \
+    "go 0x20008040;"
+
 #endif
+
 
 #elif CONFIG_SPI_BOOT
 
@@ -71,3 +152,4 @@
 #endif
 
 #endif
+
